lighthouse-core/runner.js

```diff
@@ -119,10 +119,10 @@ class Runner {
         }
       }
 
-      /** @type {Array<LH.Result.Category>} */
-      let reportCategories = [];
+      /** @type {Object<string, LH.Result.Category>} */
+      let categories = {};
       if (opts.config.categories) {
-        reportCategories = ReportScoring.scoreAllCategories(opts.config.categories, resultsById);
+        categories = ReportScoring.scoreAllCategories(opts.config.categories, resultsById);
       }
 
       const lhr = {
```

Comment:
we should probably put a `/** @type {LH.Audit.Result} */` above this so it's checked for excess properties not in the def (`reportGroups` -> `categoryGroups` isn't caught by the un-updated type in `lhr.d.ts` since `reportGroups` was optional)

Reply:
yeah good call done